ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisScreenContract.java

```diff
@@ -0,0 +1,71 @@
+package net.gini.android.vision.analysis;
+
+import android.app.Application;
+import android.content.DialogInterface;
+import android.graphics.Bitmap;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import net.gini.android.vision.GiniVisionBasePresenter;
+import net.gini.android.vision.GiniVisionBaseView;
+import net.gini.android.vision.internal.util.Size;
+
+import java.util.List;
+
+import jersey.repackaged.jsr166e.CompletableFuture;
+
+/**
+ * Created by Alpar Szotyori on 08.05.2019.
+ *
+ * Copyright (c) 2019 Gini GmbH.
+ */
+interface AnalysisScreenContract {
```

Comment:
Because this is the contract to which the View and the Presenter conform to. Putting the View and Presenter interfaces inside a Contract interface is a popular MVP convention on Android. Probably due to [Google's MVP sample](https://github.com/googlesamples/android-architecture/tree/todo-mvp/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks).

Looking for the Contract interfaces in an MVP app helps to quickly see what the respective View and Presenter are "contracted" to do.

Reply:
I meant just the name, so it would be `AnalysisScreen.View` and so on. But I didn't know this convention so it's fine for me